### PR TITLE
Locations

### PR DESCRIPTION
**Pull request by markhaur**
The PR includes following accomplishments:

- Syncing of `locations` API along with its unit tests
- Mock server implementation for sending back 10k random `trips` and `locations`

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,3 +1,4 @@
 secret.env
+postgres/gen/
 pgdata/
-metabase-data/
+metabase-data/
```
Comment (jarri-abidi) [inline .gitignore:4]: these shouldn't be in .gitignore imo

File: .gitignore [hunk 2]
```diff
@@ -1,3 +1,4 @@
 secret.env
+postgres/gen/
```
Comment (jarri-abidi) [inline .gitignore]: postgres/gen/ should not be in .gitignore
it's still source code and it helps to know what should actually be generated. 
imagine there's no generated code when you checkout the code and you try running the generate cmd. how would you know it's actually generating what should be generated? what if something in the version of the code generator that you're using causes things to be generated differently? you have no point of reference in that case if these generated source files are not committed to git.

File: cmd/lambda/main.go [hunk 3]
```diff
@@ -34,5 +34,9 @@ func handler(ctx context.Context) error {
 		return errors.Wrap(err, "could not sync trips")
 	}
 
+	if err := svc.SyncLocations(ctx); err != nil {
+		return errors.Wrap(err, "could not sync trips")
```
Comment (jarri-abidi) [inline cmd/lambda/main.go]:
```suggestion
		return errors.Wrap(err, "could not sync locations")
```

File: cmd/mock/main.go [hunk 4]
```diff
@@ -1,33 +1,43 @@
...
-		}
-defer f.Close()
-		w.Header().Add("Content-Type", "application/json")
-		_, err = io.Copy(w, f)
-		if err != nil {
-			log.Printf("could not write to response writer: %v", err)
-			w.WriteHeader(http.StatusInternalServerError)
-			return
+	mux := http.NewServeMux()
+	mux.Handle("/", NewMockServer())
```
Comment (jarri-abidi) [inline cmd/mock/main.go]: why do we need a multiplexer if there's only 1 handler @markhaur ?

File: cmd/mock/main.go [hunk 5]
```diff
@@ -1,33 +1,43 @@
...
+	}
+
+	sig := make(chan os.Signal, 1)
+	signal.Notify(sig, os.Interrupt)
+
+	go func() {
+		logger.Log("transport", "http", "address", *addr, "msg", "listening")
+		if err := server.ListenAndServe(); err != http.ErrServerClosed {
+			logger.Log("transport", "http", "address", *addr, "msg", "failed", "err", err)
+			sig <- os.Interrupt // trigger shutdown of other resources
```
Comment (jarri-abidi) [inline cmd/mock/main.go]: what other resources?

File: cmd/mock/server.go [hunk 6]
```diff
@@ -0,0 +1,47 @@
...
+
+type server struct{}
+
+func NewMockServer() http.Handler {
+	s := server{}
+
+	router := way.NewRouter()
+
+	router.Handle("GET", "/trips", s.handleListTrips())
+	router.Handle("POST", "/locations", s.handleListLocations())
```
Comment (jarri-abidi) [inline cmd/mock/server.go]: did you check the docs? is it actually GET for trips and POST for locations?

File: postgres/migrations/000001_create_tables.up.sql [hunk 7]
```diff
@@ -20,3 +20,18 @@ CREATE TABLE IF NOT EXISTS trips (
...
+   device_id VARCHAR(50) NOT NULL,
+   extra VARCHAR(50) NOT NULL,
+   edt VARCHAR(50) NOT NULL,
+   eid INTEGER NOT NULL,
+   latitude FLOAT NOT NULL,
+   longitude FLOAT NOT NULL,
+   head INTEGER NOT NULL,
+   odo FLOAT NOT NULL,
+   alt FLOAT NOT NULL
+);
```
Comment (jarri-abidi) [inline postgres/migrations/000001_create_tables.up.sql]: please add newline at end of file

**Review by jarri-abidi — CHANGES_REQUESTED** [review]
.

File: cmd/mock/server.go (hunk 6 above)
Comment (markhaur) [reply cmd/mock/server.go:24]: In our client, we're making a POST request to fetch locations. So I assumed you've already checked the doc and implemented accordingly.

File: cmd/mock/main.go (hunk 4 above)
Comment (markhaur) [reply cmd/mock/main.go]: done

**Review by jarri-abidi — APPROVED** [review]

**Pull request merged**